src/Tokenomia/Vesting/PrivateSale.hs

```diff
@@ -0,0 +1,254 @@
+    return ()
+
+jsonToPrivateSale ::
+    ( MonadIO m
+    , MonadError TokenomiaError m
+    , MonadReader Environment m
+    ) =>
+    FilePath ->
+    m PrivateSale
+-- eitherDecodeFileStrict :: FromJSON a => FilePath -> IO (Either String a)
+jsonToPrivateSale jsonFilePath = do
+    fileContents <- liftIO . readFile $ jsonFilePath
+    case (decode fileContents :: Maybe PrivateSale) of
+        Nothing -> throwError . BlockFrostError . B.BlockfrostError $ "Unable to parse JSON"
+        Just ps -> return ps
+
+getTxhsByPrivateSale :: PrivateSale -> [B.TxHash]
+getTxhsByPrivateSale ps = invTxhs
+  where
+    invTxhs :: [B.TxHash]
+    invTxhs = (^. invTx) <$> invs
+
+    invs :: [Investment]
+    invs = getPsInvestments ps
+
+verifyTxs ::
+    ( MonadIO m
+    , MonadError TokenomiaError m
+    , MonadReader Environment m
+    ) =>
+    [Investment] ->
+    [B.TxHash] ->
+    m ()
+verifyTxs invs = mapM_ (verifyTx invs)
+  where
+    verifyTx ::
+        ( MonadIO m
+        , MonadError TokenomiaError m
+        , MonadReader Environment m
+        ) =>
+        [Investment] ->
+        B.TxHash ->
+        m ()
+    verifyTx invs txh = do
+        bfTx <- getTxByTxHash txh
+        inv <- liftEither $ getInvByTxHash txh invs
+        let invTokenName = getTokenName $ inv ^. invAssetClass -- :: Text
+            invAmount' = inv ^. invAmount -- :: Integer
+            bfAmts = bfTx ^. B.outputAmount -- :: [B.Amount]
+            amt = getAmountByTokenName invTokenName bfAmts -- :: B.Amount
+        if getDiscreteAmount amt == invAmount'
+            then return ()
+            else throwError . BlockFrostError . B.BlockfrostError $ "Amounts don't match"
+
+getTreasAddrTxs ::
+    ( MonadIO m
+    , MonadError TokenomiaError m
+    , MonadReader Environment m
+    ) =>
+    PrivateSale ->
+    m [B.AddressTransaction]
+getTreasAddrTxs ps =
+    do
+        prj <- projectFromEnv''
+        liftIO $
+            B.runBlockfrost prj $ do
+                B.getAddressTransactions (ps ^. psAddress)
+        >>= \case
+            Left e -> throwError $ BlockFrostError e
+            Right res -> return res
+
+getTxByTxHash ::
+    ( MonadIO m
+    , MonadError TokenomiaError m
+    , MonadReader Environment m
+    ) =>
+    B.TxHash ->
+    m B.Transaction
+getTxByTxHash txh =
+    do
+        prj <- projectFromEnv''
+        liftIO $
+            B.runBlockfrost prj $ do
+                B.getTx txh
+        >>= \case
+            Left e -> throwError $ BlockFrostError e
+            Right res -> return res
+
+getTokenName :: AssetClass -> Text
+getTokenName = pack . unpackChars . fromBuiltin . unTokenName . snd . unAssetClass
+
+getInvByTxHash :: B.TxHash -> [Investment] -> Either TokenomiaError Investment
+getInvByTxHash txh (inv : invs) =
+    case find (\inv -> (inv ^. invTx) == txh) invs of
+        Nothing -> throwError . BlockFrostError . B.BlockfrostError $ "Investment list is empty"
```

Comment:
You could use `throwError` here, specialized to `Either`, but I'd say it's more natural to use `Left`. And the explicit pattern matching can be replaced by `maybe`